src/Cardano/Wallet/Api/Types/WalletAddressPoolGap.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Cardano.Wallet.Api.Types.WalletAddressPoolGap
+    ( WalletAddressPoolGap
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap, getAddressPoolGap, mkAddressPoolGap )
+import Data.Aeson
+    ( FromJSON, ToJSON, Value (Number), parseJSON, toJSON )
+import GHC.Generics
+    ( Generic )
+
+newtype WalletAddressPoolGap = WalletAddressPoolGap
+    { getWalletAddressPoolGap :: AddressPoolGap }
```

Comment:
You mentioned you weren't satisfied by this. What about something like a newtype `ApiT` or `InApi` to put around types that are used in the Api. That's doesn't change much the implementation in the end but it removes some trouble in finding good names.

Reply:
I think the idea to use `ApiT` (or something similar is a good one. Though perhaps it rests on the assumption that every API type has a corresponding "normal" type somewhere else? Perhaps this assumption is always valid?